runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/StreamingStepMetricsContainer.java

```diff
@@ -377,11 +385,11 @@ private void deleteStaleCounters(
    */
   @VisibleForTesting
   Iterable<PerStepNamespaceMetrics> extractPerWorkerMetricUpdates() {
-    ConcurrentHashMap<MetricName, Long> counters = new ConcurrentHashMap<MetricName, Long>();
-    ConcurrentHashMap<MetricName, Long> per_worker_gauges =
-        new ConcurrentHashMap<MetricName, Long>();
-    ConcurrentHashMap<MetricName, LockFreeHistogram.Snapshot> histograms =
-        new ConcurrentHashMap<MetricName, LockFreeHistogram.Snapshot>();
+    ImmutableMap.Builder<MetricName, Long> counters = new ImmutableMap.Builder<MetricName, Long>();
+    ImmutableMap.Builder<MetricName, Long> per_worker_gauges =
```

Comment:
how about perWorkerGaugeUpdates and similar for histogram map?
using _ isn't java naming convention and updates clarifies it is based upon counter snapshot